Update roadmap with 2025-07-04 changelog features (#3621)
neondatabase/website: content/docs/introduction/roadmap.md

## What we've shipped recently 🚢

> - **TanStack integration & new open-source tools**: Neon is now the official database partner of TanStack, with new open-source tools including Create TanStack Add-on, Neon Launchpad, NeonDB CLI + SDK, and Vite Plugin for Neon to streamline fullstack development with TanStack, Vite, and Postgres.

Reviewer: ```suggestion
- **TanStack integration & new open-source tools**: Neon is now the official database partner of TanStack, with new open-source tools including a Vite Plugin for Neon to streamline fullstack development with TanStack, Vite, and Postgres.
```

- **Neon Chat for Visual Studio Code**: This AI-powered assistant lets you chat with the latest Neon documentation without leaving your IDE. You can find it here: [Neon Postgres VS Code Extension](https://marketplace.visualstudio.com/items?itemName=buildwithlayer.neon-integration-expert-15j6N).
- **A GitHub Copilot extension**: This extension provides chat-based access to the latest Neon documentation directly from your repository. You can find it here: [Neon Postgres Copilot Extension](https://github.com/marketplace/neon-database)
- **Schema Diff API**: Neon now supports schema checks in agentic systems and deployment pipelines with the new schema diff API endpoint. Learn more about [Schema Diff](/docs/guides/schema-diff), which is also available via the console and CLI.

> - **Neon Auth (Early Access)**: Sync user profiles from your auth provider to your database automatically. Now includes OAuth provider management - enable or disable providers (Google, GitHub, Microsoft) and choose between shared Neon Auth credentials or custom client credentials. Currently in Early Access. See [Neon Auth](/docs/guides/neon-auth) for details.

Reviewer: ```suggestion
- **Neon Auth**: Sync user profiles from your auth provider to your database automatically. Includes OAuth provider management - enable or disable providers (Google, GitHub, Microsoft) and choose between shared Neon Auth credentials or custom client credentials. See [Neon Auth](/docs/guides/neon-auth) for details.
```
